ControlRoomApplication/ControlRoomApplication/GUI/MainForm.cs

```diff
@@ -872,16 +871,48 @@ private void groupBox3_Enter(object sender, EventArgs e)
         private void acceptSettings_Click(object sender, EventArgs e)
         {
             finalSettings = !finalSettings;
+
             if (finalSettings == false)
             {
 
-                //Editing text to relflect state -- When the settings are being edited, the button will need to say 'finalize'
-                acceptSettings.Text = "Edit Settings";
-                if (comboBox2.Text != "Production Weather Station")
+                if (comboBox2.Text == "Production Weather Station")
+                {
+                    
+                    lastCreatedProductionWeatherStation = BuildWeatherStation();
+                   
+                    if (lastCreatedProductionWeatherStation != null)
+                    {
+                        
+                        startButton.BackColor = System.Drawing.Color.LimeGreen;
+                        startButton.Enabled = true;
+                    }
+                    else
+                    {
+                        this.WCOMPortToolTip.Show("Could not create production weather station on this port.", label8);
+                        txtWSCOMPort.BackColor = System.Drawing.Color.Yellow;
+
+
+                    }
+
+                }
+                else
                 {
+
                     startButton.BackColor = System.Drawing.Color.LimeGreen;
                     startButton.Enabled = true;
                 }
+
+                //if there is an error with the production weather station, display a tooltip & do not let the user start the telescope
+
+
+
+                //Editing text to relflect state -- When the settings are being edited, the button will need to say 'finalize'
+                acceptSettings.Text = "Edit Settings";
+
+
+
```

Comment:
There is some excessive whitespace added to this file. It would probably be best to reduce some of it unless there is some reason for it I don't know about.

- [ ] Consider reducing whitespace